src/tcpedit/edit_packet.c

```diff
@@ -63,7 +63,9 @@ fix_ipv4_checksums(tcpedit_t *tcpedit, struct pcap_pkthdr *pkthdr, ipv4_hdr_t *i
     
 
     /* calc the L4 checksum if we have the whole packet && not a frag or first frag */
-    if (pkthdr->caplen == pkthdr->len && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
+    if (pkthdr->caplen == pkthdr->len
+            && pkthdr->len > ntohs(ip_hdr->ip_len)
+            && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
```

Comment:
The comment following `needtorecalc` is that it should be set should the packet change, so it feels right to set the flag when I edit the packet bytes or length through fuzzing.
Do you agree ?

I think I messed up that part: the test should fail if `ip_hdr->ip_len < (iphdr_->ip_hl << 2) // ie. totlen < hdrlen` (right ?)
I'll rework that patch, and also have a look at ipv6 to be coherent.

Also, I'll rebase so as to prevent merge conflicts.